neo/SmartContract/Native/NativeContract.cs

```diff
@@ -95,32 +109,32 @@ protected virtual long GetPriceForMethod(string method)
             return 0;
         }
 
-        protected virtual StackItem Main(ApplicationEngine engine, string operation, VMArray args)
-        {
-            switch (operation)
-            {
-                case "onPersist":
-                    return OnPersist(engine);
-                case "supportedStandards":
-                    return SupportedStandards.Select(p => (StackItem)p).ToList();
-            }
-            throw new NotSupportedException();
-        }
-
         internal virtual bool Initialize(ApplicationEngine engine)
         {
             if (engine.Trigger != TriggerType.Application)
                 throw new InvalidOperationException();
             return true;
         }
 
+        [ContractMethod]
+        protected StackItem OnPersist(ApplicationEngine engine, VMArray args)
+        {
+            return OnPersist(engine);
+        }
+
         protected virtual bool OnPersist(ApplicationEngine engine)
         {
             if (engine.Trigger != TriggerType.System)
                 throw new InvalidOperationException();
             return true;
         }
 
+        [ContractMethod(Name = "supportedStandards")]
```

Comment:
We can create a constructor, because is just one property

Reply:
I'm going to add more properties.

Reply:
[ContractMethod("supportedStandards", Price = 123)] ?